detectorCommon/src/xtalk_neighbors_moduleThetaMergedSegmentation.cpp

```diff
@@ -0,0 +1,224 @@
+  double dummy_xtalk_theta=0.2e-2; // type 2
+  double dummy_xtalk_diagonal=0.04e-2; // type 3
+  double dummy_xtalk_tower=0.1e-2; // type 4
+
+  // retrieve layer/module/theta of cell under study
+  int layer_id = aDecoder.get(aCellId, aFieldNames[idLayerField]);
+  //int module_id = aDecoder.get(aCellId, aFieldNames[idModuleField]);
+  int theta_id = aDecoder.get(aCellId, aFieldNames[idThetaField]);
+  
+  int layer_indice_offset = layer_id - aFieldExtremes_layer[0][idLayerField].first;
+
+  // now find crosstalk neighbours
+  dd4hep::DDSegmentation::CellID cID = aCellId;
+
+  // for neighbours across different layers, we have to take into
+  // account that merging along module and/or theta could be different
+  // so one cell in layer N could be neighbour to several in layer N+-1
+  // The cells are classified in a different way whether they are
+  // direct neighbours (common surface), diagonal neighbours (common edge or vertex)
+  // or neither.
+  // To decide this, we need to check how the cells are related in both directions:
+  // neighbours (edge at least partially in common), diagonal neigbours (common vertex),
+  // none
+  for (int deltaLayer = -1; deltaLayer<2; deltaLayer+=2) {
+    
+    // no neighbours in layer N-1 for innermost layer
+    if (layer_id == aFieldExtremes_layer[layer_indice_offset][idLayerField].first && deltaLayer<0) continue;
+    // and in layer N+1 for outermost layer
+    if (layer_id == aFieldExtremes_layer[layer_indice_offset][idLayerField].second && deltaLayer>0) continue;
+    
+    // set layer field of neighbour cell
+    aDecoder.set(cID, aSeg.fieldNameLayer(), layer_id + deltaLayer);
+    
+    // find the neighbour(s) in module and theta
+    // if the numbers of module (theta) merged cells across 2 layers are the
+    // same then we just take the same module (theta) ID
+    // otherwise, we need to do some math to account for the different mergings
+    // note: even if number of merged cells in layer-1 is larger, a cell
+    // in layer could neighbour more than one cell in layer-1 if the merged
+    // cells are not aligned, for example if cells are grouped by 3 in a layer
+    // and by 4 in the next one, cell 435 in the former (which groups together
+    // 435-436-437) will be neighbour to cells 432 and 436 of the latter
+    // this might introduce duplicates, we will remove them later
+    // another issue is that it could add spurious cells beyond the maximum module number
+    // to prevent this we would need to know the max module number in layer -1
+    // which would require modifying this function passing the extrema for all layers
+    // instead of the extrema only for a certain layer
+    // this border effect is also present in the original method..
+    for (int i=-1; i <= aSeg.mergedThetaCells(layer_id); i++) {
+      int theta_id_neighbour = (theta_id + i) - ((theta_id + i) % (aSeg.mergedThetaCells(layer_id+deltaLayer) ? aSeg.mergedThetaCells(layer_id+deltaLayer):1) );
```

Comment:
```suggestion
      double dummy_xtalk = 0.0;  // will be set to the coefficient for this neighbour
      int theta_id_neighbour = (theta_id + i) - ((theta_id + i) % (aSeg.mergedThetaCells(layer_id+deltaLayer) ? aSeg.mergedThetaCells(layer_id+deltaLayer):1) );
```
moving dummy_xtalk to the loop where it will be used.